neqo-transport/src/connection/tests/handshake.rs

```diff
@@ -479,10 +479,11 @@ fn reorder_handshake() {
     let (_, s_hs) = split_datagram(&s1.unwrap());
     assert!(s_hs.is_some());
 
-    // Pass just the handshake packet in and the client can't handle it.
+    // Pass just the handshake packet in and the client can't handle it yet.
+    // It can only send another Initial packet.
     now += RTT / 2;
-    let res = client.process(s_hs, now);
-    assert_ne!(res.callback(), Duration::new(0, 0));
+    let dgram = client.process(s_hs, now).dgram();
+    assertions::assert_initial(&dgram.as_ref().unwrap(), false);
```

Comment:
This is the one case where the new code actually gets hit.  Good to know we're getting almost enough tests to have something fail in response to every change.